cmd/livepeer/starter/starter.go

```diff
@@ -1124,7 +1124,12 @@ func StartLivepeer(ctx context.Context, cfg LivepeerConfig) {
 					panic(fmt.Errorf("'pricePerUnit' value specified for model '%v' in pipeline '%v' must be >= 0, provided %v", config.ModelID, config.Pipeline, config.PricePerUnit))
 				}
 				pricePerPixel := new(big.Rat).Quo(pricePerUnit, pixelsPerUnit)
-				autoPrice, err := core.NewAutoConvertedPrice(currency, pricePerPixel, nil)
+				var autoPrice *core.AutoConvertedPrice
+				if *cfg.Network == "offchain" {
+					autoPrice = core.NewFixedPrice(pricePerPixel)
+				} else {
+					autoPrice, err = core.NewAutoConvertedPrice(currency, pricePerPixel, nil)
+				}
```

Comment:
I believe this change should be reverted. The auto price converter will not panic if you call it with a base currency like `wei` or `ETH`. If you start the service in offchain mode but provide a custom currency price tho, that's a configuration error. We cannot convert the custom-currency price config into a wei price here, that could lead to even worse problems (e.g. charging 1 wei per pixel when the operator wanted 1 USD).

By reverting this, we will keep the behavior of failing startup in case that wrong configuration is made. The operator should then either fix the price to not be in a custom currency, or configure the node to be onchain.

Reply:
This was added because the NewAutoConvertedPrice does not work without a priceFeed being loaded and in `offChain` mode it will not be loaded. I will try to ensure that pricing in the AI case is only used in on-chain mode 👍🏻.

Reply:
Fixed in https://github.com/livepeer/go-livepeer/pull/3103/commits/8e654d7ad60b68b609be069ebcdfd045978102f6.

Reply:
I'm ok with the current solution. Just to clarify this:

> This was added because the NewAutoConvertedPrice does not work without a priceFeed being loaded and in offChain mode it will not be loaded.

It doesn't fail if no custom currency is specified (wei or eth): https://github.com/livepeer/go-livepeer/blob/706ec33286344613ec37c4f0ac4d4f3d17657049/core/autoconvertedprice.go#L48

So what is actually wrong is configuring `network=offchain` and a custom currency, since there's no way to use a custom currency offchain. My argument was that that is a config error, not a problem with this code. I'm fine with the current behavior you've implemented tho, just not setting any price anywhere when offchain.